sdks/python/apache_beam/coders/union_coder_test.py

```diff
@@ -0,0 +1,61 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# pytype: skip-file
+
+import logging
+import unittest
+
+from apache_beam.coders import coders
+from apache_beam.coders.union_coder import UnionCoder
+from apache_beam.coders.avro_record import AvroRecord
+
+
+class AvroTestCoder(coders.AvroGenericCoder):
+  SCHEMA = """
+  {
+    "type": "record", "name": "test",
+    "fields": [
+      {"name": "name", "type": "string"},
+      {"name": "age", "type": "int"}
+    ]
+  }
+  """
+
+  def __init__(self):
+    super().__init__(self.SCHEMA)
+
+
+class UnionCoderTest(unittest.TestCase):
+  def test_basics(self):
+
+    coder = UnionCoder([AvroTestCoder(), coders.BooleanCoder()])
+
+    self.assertEqual(coder.is_deterministic(), False)
```

Comment:
Not obvious to me that why `is_deterministic()` gives false -- checked that this is because "AvroTestCoder" is false. (there is even a TODO(#19628) in AvroGenericCoder suggesting deterministic or not is not confirmed)

If test deterministic we'd better have an independent test that feed both deterministic and non-deterministic test coders as test cases

Reply:
This mainly tests whether the `is_deterministic` API works.